pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java

```diff
@@ -80,6 +80,18 @@ public void testFieldSpec() {
     Assert.assertEquals(fieldSpec1.hashCode(), fieldSpec2.hashCode());
     Assert.assertEquals(fieldSpec1.getDefaultNullValue(), "false");
 
+    // Single-value boolean type dimension field with max length and default null value.
+    fieldSpec1 = new DimensionFieldSpec();
+    fieldSpec1.setName("svDimension");
+    fieldSpec1.setDataType(BOOLEAN);
+    fieldSpec1.setDefaultNullValue(false);
+    fieldSpec1.setMaxLength(20000);
```

Comment:
Why do fixed length data types (BOOLEAN/int/float/etc) need maxLength? If not, then we should not allow setting max-length for those and only apply to variable length data types (String/Bytes).

Reply:
Good point. The `maxLength` field only applies to STRING field right now (not even BYTES)

Reply:
Thanks @mayankshriv  @Jackie-Jiang for review. I will fix this test. Additional to that, do you suggest me to enforce configurable maxLength on String fields only? 
Is it possible that Pinot will support maxLength on other variable length data types in the future?

In our use case, we only allow user to set maxLength on String columns.

Reply:
@Jackie-Jiang How about multi-valued fixed length data types? E.g. Does maxLength applied to multi-value long fields?

Reply:
@deemoliu The maxLength should only be applied to String field. The method names in Schema.java should reflect this as well. i.e., the method name should be addStringMetric().

Reply:
@chenboat @Jackie-Jiang thanks for the code review. I added precondition in Schema.java for dimension columns. Regarding metrics column, does METRIC column support String column? 

I think it should be supported according to https://docs.pinot.apache.org/configuration-reference/schema#metricfieldspecs 

However in the schema.java it weren't been supported.
https://github.com/apache/incubator-pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java#L447

Reply:
It looks like the doc is not in sync with the code. The code is usually the source of truth. For this PR, I think the update to METRICS column can be skipped.

Reply:
I see. Then I will remove the code related to "metrics with max length" (since maxlength ony applied to String and metrics columns cannot be string type) and only keep "dimension with max length".